### PR TITLE
Check to limit nr of threads for small problems

### PR DESCRIPTION
**Pull request by bettinaheim**

### CODE REVIEW

**Review by swernli — APPROVED** [review]
Changes seem good to me. Have we verified that these boundaries don't make the perf of small problems on Windows too much worse than it was before the original improvements get checked in?

File: src/Simulation/Native/src/external/fused.hpp [hunk 1]
```diff
@@ -145,7 +145,10 @@ class Fused
             if (envNT == NULL) { // If the user didn't force the number of threads, make an intelligent guess
                 int nMaxThrds = std::thread::hardware_concurrency();        // Logical HW threads
                 if (nMaxThrds > 4) nMaxThrds/= 2;                           // Assume we have hyperthreading (no consistent/concise way to do this)
-                if (wfnCapacity < 1u << 20) {
+                if (wfnCapacity < 1ul << 14)      nMaxThrds = 1;
+                else if (wfnCapacity < 1ul << 16) nMaxThrds = 2;
```
Comment (rmshaffer) [inline src/Simulation/Native/src/external/fused.hpp:149]: Minor note: looks like this line would actually _increase_ `nMaxThrds` from 1 to 2 if running on a single-core CPU?

Comment (dbwz8) [reply src/Simulation/Native/src/external/fused.hpp:149]: No, but I haven't seen a 1 CPU machine in a long time. Easy to add the check if desired.

Comment (rmshaffer) [reply src/Simulation/Native/src/external/fused.hpp:149]: Up to you. As you say, this is probably a rare case in practice.

**Pull request merged**